sycl/test/extensions/annotated_usm/invalid_usm_kind.cpp

```diff
@@ -0,0 +1,113 @@
+  TEST(malloc_host_annotated, N, q, properties{usm_kind_shared, alignment<4>});
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Input property list contains conflicting USM kind.}}
+  // expected-error@+1 {{no matching function for call to 'malloc_host_annotated'}}
+  TEST(malloc_host_annotated, N, Ctx, properties{usm_kind_device});
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Input property list contains conflicting USM kind.}}
+  // expected-error@+1 {{no matching function for call to 'malloc_host_annotated'}}
+  TEST((malloc_host_annotated<int>), N, q, properties{buffer_location<1>, usm_kind_shared});
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Input property list contains conflicting USM kind.}}
+  // expected-error@+1 {{no matching function for call to 'malloc_host_annotated'}}
+  TEST((malloc_host_annotated<int>), N, Ctx, properties{buffer_location<2>, usm_kind_device});
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Input property list contains conflicting USM kind.}}
+  // expected-error@+1 {{no matching function for call to 'malloc_device_annotated'}}
+  TEST(malloc_device_annotated, N, q, properties{usm_kind_host});
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Input property list contains conflicting USM kind.}}
+  // expected-error@+1 {{no matching function for call to 'malloc_device_annotated'}}
+  TEST(malloc_device_annotated, N, dev, Ctx, properties{usm_kind_shared});
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Input property list contains conflicting USM kind.}}
+  // expected-error@+1 {{no matching function for call to 'malloc_device_annotated'}}
+  TEST((malloc_device_annotated<int>), N, q, properties{buffer_location<1>, usm_kind_host});
+
+  // expected-error-re@sycl/ext/oneapi/experimental/annotated_usm/alloc_util.hpp:* {{static assertion failed due to requirement {{.+}}: Input property list contains conflicting USM kind.}}
+  // expected-error@+1 {{no matching function for call to 'malloc_device_annotated'}}
+  TEST((malloc_device_annotated<int>), N, dev, Ctx, properties{buffer_location<2>, usm_kind_shared});
+}
+
+// Duplicated properties (consistent or conflicting) exist in the property list
+void testInvalidPropList() {
+  // expected-error-re@sycl/ext/oneapi/properties/properties.hpp:* {{static assertion failed due to requirement {{.+}}: Duplicate properties in property list.}}
+  properties InvalidPropList{usm_kind_device, usm_kind_host};
```

Comment:
Can you add the following test case:

```
properties InvalidPropList{usm_kind_device, usm_kind_device};
```

Reply:
This is determined by the implementation of the `properties` class

Reply:
Oh, I was misremembering the properties extension. I see that it actually specifies that duplicate properties can't be in the same property list:

"In the same way that two different runtime properties of the same type cannot be applied to the same object, two compile-time constant property values of the same property class T cannot belong to the same properties, whether the property value is the same or different."